### PR TITLE
WIP: Add inline example for pygmt.datasets.list_sample_data

### PR DESCRIPTION
**Pull request by maxrjones**
**Description of proposed changes**

This PR adds an inline example for pygmt.datasets.list_sample_data. It is a possible solution to https://github.com/GenericMappingTools/pygmt/issues/1774 and would be easier to maintain than table/list in the docstring since the output from the example could be easily updated.

It is listed as WIP because the style checks fail with line-too-long errors.

Addresses #1774 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/samples.py [hunk 1]
```diff
@@ -20,6 +20,19 @@ def list_sample_data():
...
+    --------
+    >>> import pygmt
+    >>> pygmt.datasets.list_sample_data()
+    {'bathymetry': 'Table of ship bathymetric observations off Baja California',
+    'fractures': 'Table of hypothetical fracture lengths and azimuths',
+    'hotspots': 'Table of locations, names, and symbol sizes of hotpots from  Mueller et al., 1993',
+    'japan_quakes': 'Table of earthquakes around Japan from NOAA NGDC database',
+    'mars_shape': 'Table of topographic signature of the hemispheric dichotomy of  Mars from Smith and Zuber (1996)',
+    'ocean_ridge_points': 'Table of ocean ridge points for the entire world',
+    'usgs_quakes': 'Table of global earthquakes from the USGS'}
```
Comment (willschlitzer) [inline pygmt/datasets/samples.py:35]: Should this list only 2-3 dataset names, and maybe use `...` to indicate that this is an incomplete list? I don't want users to think the output in the example is the complete list.
```
   {'bathymetry': 'Table of ship bathymetric observations off Baja California',
    'fractures': 'Table of hypothetical fracture lengths and azimuths',
   ...
    'usgs_quakes': 'Table of global earthquakes from the USGS'}
```

Comment (seisman) [reply pygmt/datasets/samples.py:35]: See https://github.com/GenericMappingTools/pygmt/issues/1774. The idea of this PR is to better document the available sample data.

Comment (meghanrjones) [reply pygmt/datasets/samples.py:35]: Are you proposing deprecating list_sample_data?

Comment (seisman) [reply pygmt/datasets/samples.py:35]: No, `list_sample_data` is still useful to show the available datasets in interactive presentations.

**Comment by maxrjones** [comment]
Superceded by https://github.com/GenericMappingTools/pygmt/pull/2342

**Pull request closed without merging**